Helper/Cart.php

```diff
@@ -2877,7 +2878,9 @@ public function getSaleRuleDiscounts($quote)
      */
     public function isCollectDiscountsByPlugin($quote)
     {
-        return $this->configHelper->isActive($quote->getStore()->getId())
-            && version_compare($this->configHelper->getStoreVersion(), '2.3.4', '<');
+        //By default this feature switch is disabled.
+        return $this->deciderHelper->isCollectDiscountsByPlugin()
+            || ($this->configHelper->isActive($quote->getStore()->getId())
```

Comment:
@dumega should it be `&&` instead of `||` here?
```suggestion
            && ($this->configHelper->isActive($quote->getStore()->getId())
```

Reply:
@danac-gs No, it should be `||` , 
1. If $this->deciderHelper->isCollectDiscountsByPlugin() return true, we should always collect the discounts via plugin methods.
2. If $this->deciderHelper->isCollectDiscountsByPlugin() return false, we reply on the conditional statements
```
$this->configHelper->isActive($quote->getStore()->getId())
            && version_compare($this->configHelper->getStoreVersion(), '2.3.4', '<')
```
and only if the both two statements return true the discounts would be collected via plugin.